### DISCUSSION TITLE
Need ability to change data source for whole environment

### DISCUSSION THREAD

**Discussion by nabelekt** [grafana/grafana#38280] [Dashboards, Panels & Visualizations]
Recently, the lack of ability to set the data source above the panel level has been an obstacle for us. We use different databases for test and operations/production, and the way things work now, we have to maintain separate copies of dashboards and try to keep the only difference between them to be the data source. Doing a find and replace in the dashboard json can help somewhat but is not a user-friendly solution. We are looking for the ability to set the data source with one drop-down menu or variable change on the dashboard. Within the panels, the data source list could include a "dashboard data source" option with the ability to override on a panel by panel basis. Beyond that, the ability to change the data source for all dashboards at once would also be a very big help. Could work in the same way and just trickle down to dashboards and then individual panels.

**Comment by valdemarpavesi** [comment]
hello,

you can create a variable with type=datasource

and apply the variable to all pannel's datasource selection.

regards!
Valdemar

**Comment by nabelekt** [comment]
Great, thanks! I'll try that out.

Will leave this open for now to provide the ability to share a single data source across dashboards.

**Comment by ivanahuckova** [comment]
You can do this trough data source variables. Here is the [demo of this feature on grafana.play](https://play.grafana.org/d/5eue-XDGz/datasource-variable?orgId=1) and[ here is the documentation](https://grafana.com/docs/grafana/latest/variables/variable-types/add-data-source-variable/).

**Comment by nabelekt** [comment]
Sure, but that is only within the dashboard. I am looking for a way to do it across dashboards for all dashboards in Grafana.

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
